apps/prairielearn/src/tests/questionSharing.test.ts

```diff
@@ -458,5 +484,91 @@ describe('Question Sharing', function () {
       );
       assert(publiclySharedQuestionRes.ok);
     });
+
+    step('Fail to sync if shared question is renamed', async () => {
+      const questionPath = path.join(sharingCourse.path, 'questions', SHARING_QUESTION_QID);
+      const questionTempPath = questionPath + '_temp';
+      await fs.rename(questionPath, questionTempPath);
+      await syncFromDisk.syncOrCreateDiskToSql(sharingCourse.path, logger);
+      const question_id = await sqldb.queryOptionalRow(
+        sql.get_question_id,
+        {
+          course_id: sharingCourse.id,
+          qid: SHARING_QUESTION_QID,
+        },
+        IdSchema,
+      );
+      assert(
+        question_id !== null,
+        'Sync of consuming course should not allow renaming a shared question.',
+      );
+      await fs.rename(questionTempPath, questionPath);
+    });
+
+    step('Sync through the sync page to extablish commit hash.', async () => {
+      await sqldb.queryAsync(sql.update_course_repository, {
+        course_path: sharingCourseLiveDir,
+        course_repository: sharingCourseOriginDir,
+      });
+
+      const syncUrl = `${baseUrl}/course/${sharingCourse.id}/course_admin/syncs`;
+      const response = await fetchCheerio(syncUrl);
+
+      const token = response.$('#test_csrf_token').text();
+      await fetch(syncUrl, {
+        method: 'POST',
+        body: new URLSearchParams({
+          __action: 'pull',
+          __csrf_token: token,
+        }),
+      });
+      const result = await sqldb.queryOneRowAsync(sql.select_last_job_sequence, []);
+      const job_sequence_id = result.rows[0].id;
+
+      await helperServer.waitForJobSequenceStatus(job_sequence_id, 'Success');
+    });
+
+    step('Rename shared question in origin, ensure live does not sync it', async () => {
+      const questionPath = path.join(sharingCourseOriginDir, 'questions', SHARING_QUESTION_QID);
+      const questionTempPath = questionPath + '_temp';
+      await fs.rename(questionPath, questionTempPath);
+      await execa('git', ['add', '-A'], gitOptions);
+      await execa('git', ['commit', '-m', 'rename shared question'], gitOptions);
+
+      sharingCourse = await selectCourseById(sharingCourse.id);
```

Comment:
Presumably you're doing this to pick up the updated commit hash, is that correct? IT wasn't immediately obvious to me why you didn't just use the existing `sharingCourse` here.

Reply:
Ya that's right. Should I just call `getCourseCommitHash` instead?

Reply:
I did that instead, works great and much more readable.